packages/apollo-server-fastify/src/ApolloServer.ts

```diff
@@ -0,0 +1,137 @@
+import * as fastify from 'fastify';
+import { FastifyInstance } from 'fastify';
+const { parseAll } = require('accept');
+import {
+  renderPlaygroundPage,
+  RenderPageOptions as PlaygroundRenderPageOptions,
+} from '@apollographql/graphql-playground-html';
+
+import { graphqlFastify } from './fastifyApollo';
+
+export { GraphQLOptions, GraphQLExtension } from 'apollo-server-core';
+import { ApolloServerBase, GraphQLOptions } from 'apollo-server-core';
+import { IncomingMessage, OutgoingMessage, ServerResponse } from 'http';
+
+export class ApolloServer extends ApolloServerBase {
+  async createGraphQLServerOptions(
+    request: fastify.FastifyRequest<IncomingMessage>,
+    reply: fastify.FastifyReply<OutgoingMessage>,
+  ): Promise<GraphQLOptions> {
+    const options = await super.graphQLServerOptions({ request, reply });
+    return options;
+  }
+
+  protected supportsSubscriptions(): boolean {
+    return true;
+  }
+
+  protected supportsUploads(): boolean {
+    return true;
+  }
+
+  public async applyMiddleware({
+    app,
+    cors,
+    path,
+    route,
+    disableHealthCheck,
+    onHealthCheck,
+  }: ServerRegistration) {
+    if (!path) path = '/graphql';
+
+    await app.addHook(
+      'onRequest',
+      async function(
+        this: any,
+        request: any,
+        response: ServerResponse,
+        next: any,
+      ) {
+        if (request.path !== path) {
+          return next();
+        }
+
+        if (this.playgroundOptions && request.method === 'get') {
+          // perform more expensive content-type check only if necessary
+          const accept = parseAll(request.headers);
+          const types = accept.mediaTypes as string[];
+          const prefersHTML =
+            types.find(
+              (x: string) => x === 'text/html' || x === 'application/json',
+            ) === 'text/html';
+
+          if (prefersHTML) {
+            const playgroundRenderPageOptions: PlaygroundRenderPageOptions = {
+              endpoint: path,
+              subscriptionEndpoint: this.subscriptionsPath,
+              version: this.playgroundVersion,
+              ...this.playgroundOptions,
+            };
+
+            response.setHeader('Content-Type', 'text/html');
+            return response.end(
+              renderPlaygroundPage(playgroundRenderPageOptions),
+            );
+          }
+        }
+        return next();
+      }.bind(this),
```

Comment:
I would recommend doing this in a `beforeHandler`  hook instead.

Reply:
Done. Added a `preHandler`